simphony_mayavi/sources/slim_cuds_source.py

```diff
@@ -0,0 +1,232 @@
+        self.update()
+
+    def _point_vectors_name_changed(self, value):
+        super(CUDSSource, self)._point_vectors_name_changed(value)
+        self.update()
+
+    def _cell_scalars_name_changed(self, value):
+        super(CUDSSource, self)._cell_scalars_name_changed(value)
+        self.update()
+
+    def _cell_vectors_name_changed(self, value):
+        super(CUDSSource, self)._cell_vectors_name_changed(value)
+        self.update()
+
+    def _set_cuds(self, cuds):
+
+        # Before refreshing the VTK CUDS object, we need to set the content
+        # of the available data in the cuds. Normally this is done by the
+        # VTKDataSource _from_ the VTK data, but our VTK data will not contain
+        # all the keys. We are therefore required to update them here.
+        all_lists = (
+            self._point_scalars_list,
+            self._point_vectors_list,
+            self._cell_scalars_list,
+            self._cell_vectors_list)
+
+        for lst, keys in zip(all_lists, _available_keys(cuds)):
+            lst[:] = sorted([key.name for key in keys])
+
+        super(SlimCUDSSource, self)._set_cuds(cuds)
+
+    def _update_vtk_cuds_from_cuds(self, cuds):
+        """Update _vtk_cuds, but limit the extraction to only the requested
+        data."""
+
+        # Extract the requested data we want.
+        points_keys = [CUBA[x]
+                       for x in [self.point_scalars_name,
+                                 self.point_vectors_name]]
+
+        cell_keys = [CUBA[x]
+                     for x in [self.cell_scalars_name,
+                               self.cell_vectors_name]]
+
+        if isinstance(cuds, (VTKMesh, VTKParticles, VTKLattice)):
+            vtk_cuds = cuds
+        else:
+            if isinstance(cuds, (ABCMesh, H5Mesh)):
+                vtk_cuds = VTKMesh.from_mesh(cuds, points_keys, cell_keys)
+            elif isinstance(cuds, ABCParticles):
+                vtk_cuds = VTKParticles.from_particles(
+                    cuds,
+                    points_keys,
+                    cell_keys)
+            elif isinstance(cuds, ABCLattice):
+                vtk_cuds = VTKLattice.from_lattice(
+                    cuds,
+                    points_keys)
+            else:
+                msg = 'Provided object {} is not of any known cuds type'
+                raise TraitError(msg.format(type(cuds)))
+        self._vtk_cuds = vtk_cuds
+
+    def _update_data(self):
+        # We need to silence the behavior of the VTKDataSource, otherwise it
+        # will overwrite the point_scalar_list etc. with data from the vtk
+        # datasource, which in this case contains only the data we care
+        # about.
+        pass
+
+
+def _to_cuba(iterable):
+    """Small generator that processes an iterable of strings
+    into an iterable of CUBA keys"""
+    for elem in iterable:
+        try:
+            yield CUBA[elem]
+        except ValueError:
+            pass
+
+
+def _available_keys(cuds):
```

Comment:
utility function do not have to be private. However about moving them to the core module